### PR TITLE
feat: prevent a workflow from stopping when pre-commit fails

### PR DESCRIPTION
**Pull request by TomohitoAndo**
Signed-off-by: Tomohito Ando <tomohito.ando@tier4.jp>

## Description

Fixed workflow stopping when it failed.
For example, if there are clang-format failure and cpplint failure, workflow stops without fixing clang-format errors.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TomohitoAndo** [comment]
I confirmed this works. (private repository in TIER IV)

https://github.com/tier4/gsm8_interface/actions/runs/3508437414/jobs/5876839435

File: pre-commit/action.yaml [hunk 1]
```diff
@@ -18,8 +18,23 @@ runs:
...
+        python -m pip install pre-commit
+      shell: bash
+
     - name: Run pre-commit
-      uses: pre-commit/action@v2.0.3
-      with:
-        extra_args: --all-files --config ${{ inputs.pre-commit-config }}
-        token: ${{ inputs.token }}
+      run: |
+        pre-commit run -a
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:28]:
```suggestion
        pre-commit run -a --config ${{ inputs.pre-commit-config }}
```

File: pre-commit/action.yaml [hunk 2]
```diff
@@ -18,8 +18,23 @@ runs:
...
+      shell: bash
+
+    - name: Push pre-commit fixes
+      # Prevent a workflow from stopping when pre-commit fails
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:32]: [IMO] I prefer this comment.

```suggestion
      # Push changes even when pre-commit fails
```

File: pre-commit/action.yaml [hunk 3]
```diff
@@ -18,8 +18,23 @@ runs:
...
+      if: always()
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:33]: [IMO] I prefer this comment.

```suggestion
      if: always() # Push changes even when pre-commit fails
```

**Review by kenji-miyake — APPROVED** [review]
Great! Thank you. :pray:

**Pull request merged**